src/librustc_mir/transform/promote_consts.rs

```diff
@@ -360,8 +369,6 @@ impl<'tcx> Validator<'_, 'tcx> {
                 }
             },
             Candidate::Argument { bb, index } => {
-                assert!(self.explicit);
-
                 let terminator = self.body[bb].terminator();
                 match &terminator.kind {
                     TerminatorKind::Call { args, .. } => {
```

Comment:
I guess you didn't want to emit the error here?

Reply:
I can imagine calling `validate_candidate` and not wanting to emit errors, e.g. for diagnostics.